internal/provider/resource_cloudflare_record.go

```diff
@@ -49,7 +49,7 @@ func resourceCloudflareRecordCreate(ctx context.Context, d *schema.ResourceData,
 		ZoneID: d.Get("zone_id").(string),
 	}
 
-	proxied, proxiedOk := d.GetOkExists("proxied")
```

Comment:
i think you'll want to leave this to handle the unset, true and false conditions correctly.